pinot-spi/src/main/java/org/apache/pinot/spi/annotations/segment/lifecycle/SegmentDeletionListener.java

```diff
@@ -0,0 +1,32 @@
+
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.annotations.segment.lifecycle;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface SegmentDeletionListener {
```

Comment:
(optional) This can be an inner interface in `SegmentDeletionManager`

Reply:
@Jackie-Jiang can you elaborate why? In fact, I would have thought that the methods in `PinotSegmentDeletionListener` interface should be declared here instead. Why split it into two interfaces?

Reply:
@mcvsubbu You are right. We should put the interfaces together and ideally put them into `spi` so that people only need to pull `spi` dependency when adding plug-ins. That is not possible right now because `HelixManager` is not included in `spi`. We can probably move them into `common` as of now.
@saurabhd336 wdyt?

Reply:
Put the interface in spi. 
Implementatins can be inside. The init method need not be a part of the interface. Each implementtion can initialize whatever they want.

Reply:
Actually `init()` has to be part of the interface, or the caller (`PinotHelixResourceManager`) won't be able to call it

Reply:
I think the pattern we follow is that the object gets initialized in the starter, no? And then gets passed down to whoever needed.

Reply:
I was using some other SPI annotation classes as reference.. (like https://github.com/apache/pinot/blob/master/pinot-spi/src/main/java/org/apache/pinot/spi/annotations/minion/TaskExecutorFactory.java)
where annotation is defined in spi while interface is in the relevant module.

@mcvsubbu I kind of needed PinotHelixResourceManager in the listener classes to be able to access whatever metadata individual listener might need for the segment / table being deleted. I can check if moving both the annotation and interface inside spi would work.. Since we're initialzing with reflections, having custom constructors could be a challenge. Will try to check if we can move to initialising without reflections.